### ISSUE TITLE
Adding workflow_call to a workflow fails lint

### ISSUE THREAD

**Issue by BretFisher** [rhysd/actionlint#87]
If I add `workflow_call` to a simple workflow, I'll get an error about an env later in the file.

### This passes actionlint 1.6.8:

```yaml
---
name: Lint Code Base

on:
  push:

jobs:
  build:
    name: Lint Code Base
    runs-on: ubuntu-latest

    steps:
      - name: Checkout Code
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Lint Code Base
        uses: github/super-linter@v4
        env:
          VALIDATE_ALL_CODEBASE: false
          DEFAULT_BRANCH: master
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### This fails (only added one line `workflow_call`)

```yaml
---
name: Lint Code Base

on:
  push:
  workflow_call:

jobs:
  build:
    name: Lint Code Base
    runs-on: ubuntu-latest

    steps:
      - name: Checkout Code
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Lint Code Base
        uses: github/super-linter@v4
        env:
          VALIDATE_ALL_CODEBASE: false
          DEFAULT_BRANCH: master
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### error:

```text
 property "github_token" is not defined in object type {} [expression]
   |
24 |           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   |                             ^~~~~~~~~~~~~~~~~~~~
```

**Comment by ferrarimarco** [comment]
I saw this happen with actionlint v1.6.8 as well.

**Comment by BeeeQueue** [comment]
This is because when you use `workflow_call` you need to [specify what secrets you want](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow) passed into it.

The linter checks for this and verifies that any secret used is specified in the `secrets` config - but it doesn't know that `GITHUB_TOKEN` is automatically added by GH so it thinks it is missing.

**Comment by ferrarimarco** [comment]
This shouldn't be the case for `secrets.GITHUB_TOKEN` and `github.token`, right?

From the page you linked:

> When a reusable workflow is triggered by a caller workflow, the github context is always associated with the caller workflow. The called workflow is automatically granted access to github.token and secrets.GITHUB_TOKEN. For more information about the github context, see "Context and expression syntax for GitHub Actions."

**Comment by BeeeQueue** [comment]
Indeed. I updated the comment to make it clearer

**Comment by mkj-is** [comment]
I confirm this happens due to missing exception for the automatically supplied `GITHUB_TOKEN` secret.

I was searching for a way how to fix it, but I am no expert in Go. Do you think adding the secret manually to this map is a good idea? Is there another way?

https://github.com/rhysd/actionlint/blob/main/parse.go#L488

**Comment by FalconerTC** [comment]
Any suggestions here? Still seeing this error

**Comment by powerman** [comment]
Same here. Looks like it wasn't properly fixed or there is a regression in latest version(s). Please reopen the issue.
